### DISCUSSION TITLE
[4.x] Add stack trace to warnings

### DISCUSSION THREAD

**Discussion by jordikroon** [craftcms/cms#12253] [General]
Since Craft 4, 2 warnings were added to the source and kinda became mandatory to fix (which is a good thing) because these are triggered on every request.

 - `Craft::warning('Element query executed before Craft is fully initialized.', __METHOD__);`
 - `Craft::warning('Twig instantiated before Craft is fully initialized.', __METHOD__);`

Whenever these are triggered by accident or because they were already in the code before the Craft 4 upgrade, our Sentry logs receive a lot of events over a short timespan.

Would it be possible to include the stacktrace for these warnings? Our solution for now is not ideal.
We edit the Craft source and add the following so that our Sentry presents us the Stacktrace that leads us to the cause of the issue. Which can be a module, but can also be a plugin.

```php
try {
    throw new \Exception('Element query executed before Craft is fully initialized.');
} catch (\Exception $e) {
   \Sentry\captureException($e);
}
```

**Comment by timkelty** [comment]
@jordikroon To force Craft/Yii to log stack traces for non-exceptions, you can set `traceLevel` to a non-zero integer in the `log` component of your `app.php` config:

```php
<?php

return [
    'components' => [
        'log' => [
            'traceLevel' => 5,
        ]
    ]
];
```

**Comment by AugustMiller** [comment]
Tim’s example is now part of the [Logging page](https://craftcms.com/docs/4.x/logging.html#trace-level)!

I also added a bit up front about [what gets logged](https://craftcms.com/docs/4.x/logging.html#what-gets-logged), because that (and the different levels) seemed kind of important. 😜
